packages/neo-one-server-plugin-wallet/src/bootstrap.js

```diff
@@ -0,0 +1,287 @@
+  const { network: networkName } = options;
+  if (networkName != null && typeof networkName === 'string') {
+    return networkName;
+  }
+
+  const { network } = await cli.getSession(networkConstants.PLUGIN);
+  if (network != null && typeof network === 'string') {
+    return network;
+  }
+
+  throw new Error(
+    'Bootstrap requires a network. Activate a network by ' +
+      'running `activate network <name>` or specify the network via ' +
+      '`--network <name>`',
+  );
+};
+
+async function getWallet(
+  walletName: string,
+  networkName: string,
+  cli: InteractiveCLI,
+  plugin: WalletPlugin,
+): Promise<Wallet> {
+  const wallet = await plugin.walletResourceType.getResource({
+    name: constants.makeWallet({
+      network: networkName,
+      name: walletName,
+    }),
+    client: cli.client,
+    options: {},
+  });
+
+  if (wallet == null) {
+    throw new Error(`Failed to find wallet, ${walletName}`);
+  }
+
+  return wallet;
+}
+
+async function createWallet(
+  walletName: string,
+  networkName: string,
+  cli: InteractiveCLI,
+  keystore: LocalKeyStore,
+): Promise<void> {
+  const walletResource = await cli.client.createResource({
+    plugin: constants.PLUGIN,
+    resourceType: constants.WALLET_RESOURCE_TYPE,
+    name: constants.makeWallet({ name: walletName, network: networkName }),
+    options: {
+      network: networkName,
+    },
+    cancel$: _of(),
+  });
+
+  const wallet = ((walletResource: $FlowFixMe): Wallet);
+  if (wallet == null) {
+    throw new Error(`Failed to find wallet, ${walletName}`);
+  }
+  if (wallet.wif == null) {
+    throw new Error(`Something went wrong, wif is null for ${walletName}`);
+  }
+
+  await keystore.addAccount({
+    network: networkName,
+    name: wallet.name,
+    privateKey: wifToPrivateKey(wallet.wif),
+  });
+}
+
+function getNumWallets(options): number {
+  const { wallets } = options;
+  if (wallets != null && typeof wallets === 'number') {
+    return wallets;
+  }
+  return DEFAULT_NUM_WALLETS;
+}
+
+function randomInt(max: number): number {
+  return Math.floor(Math.random() * Math.floor(max)) + 1;
+}
+
+function randomIntDist(seed: number): number {
+  if (seed < 40) {
+    return randomInt(100);
+  } else if (seed >= 40 && seed < 90) {
+    return randomInt(1000);
+  } else if (seed >= 90 && seed < 97) {
+    return randomInt(10000);
+  } else if (seed >= 97 && seed < 100) {
+    return randomInt(100000);
+  }
+  return randomInt(1000000);
+}
+
+async function createTransfers(
+  walletName: string,
+  networkName: string,
+  cli: InteractiveCLI,
+  plugin: WalletPlugin,
+  from?: ?Wallet,
+): Promise<Array<Transfer>> {
+  const wallet = await getWallet(walletName, networkName, cli, plugin);
+
+  let neo;
+  let gas;
+  if (from == null) {
+    const seed = randomInt(100);
+    neo = randomIntDist(seed);
+    gas = randomIntDist(seed);
+  } else {
+    const transferPercent = randomInt(75) / 100;
+    neo = Math.ceil(transferPercent * Number(from.neoBalance));
```

Comment:
The problem is that `from.neoBalance` is updated via polling, whereas `confirmed` is updated instantaneously when the block is confirmed. So there's a race condition where the `client` knows the block has been confirmed and the neo-one server actually updates the balance. Instead, use `client.read(<networkName>).getAccount` to get the current balance directly from the blockchain rather than the neo-one server.